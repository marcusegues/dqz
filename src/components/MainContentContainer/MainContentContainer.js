// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import type { Children } from '../../types/generalTypes';

type MainContentContainerProps = {
  children: Children,
};

export const MainContentContainer = ({
  children,
}: MainContentContainerProps) => (
  <View
    style={{
      flex: 1,
      height: '100%',
      marginHorizontal: 16,
      flexDirection: 'column',
      alignItems: 'center',
    }}
  >
    {children}
  </View>
);
