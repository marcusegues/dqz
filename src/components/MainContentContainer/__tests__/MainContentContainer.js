import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import { shallow } from 'enzyme';
import { MainContentContainer } from '../MainContentContainer';

describe('MainContentContainer', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<MainContentContainer />).toJSON();
    expect(component).toMatchSnapshot();
  });
  test('renders children according to snapshot', () => {
    const component = shallow(
      <MainContentContainer>
        <Text>Mock children</Text>
      </MainContentContainer>
    );
    expect(component.dive()).toMatchSnapshot();
    expect(
      component.containsMatchingElement(<Text>Mock children</Text>)
    ).toBeTruthy();
  });
});
