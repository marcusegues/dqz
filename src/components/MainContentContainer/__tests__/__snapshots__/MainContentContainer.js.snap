// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentContainer renders according to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "marginHorizontal": 16,
    }
  }
/>
`;

exports[`MainContentContainer renders children according to snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "height": "100%",
        "marginHorizontal": 16,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Mock children
    </Text>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Mock children
      </Text>,
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "height": "100%",
        "marginHorizontal": 16,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Mock children",
        "ellipsizeMode": "tail",
      },
      "ref": null,
      "rendered": "Mock children",
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Mock children
        </Text>,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "height": "100%",
          "marginHorizontal": 16,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Mock children",
          "ellipsizeMode": "tail",
        },
        "ref": null,
        "rendered": "Mock children",
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
